packages/core/src/config.ts

```diff
@@ -54,10 +54,19 @@ export interface Config {
    * @default 'src/App.tsx'
    */
   appShell: `${string}.tsx`;
+  /**
+   * Controls how Crackle generates `.d.ts` files.
+   *
+   * - 'bundle' rolls up all `.d.ts` files into a single file
+   * - 'declaration' generates `.d.ts` files for each file in the source directory, for maximum compatibility with TypeScript
```

Comment:
The `bundle` option seems obvious to me as to, but the `declaration` option feels a bit vague, and doesn't immediately make it obvious as to what it does. Maybe something like `preserve`, or maybe `preserveModules`, would fit better, especially given that it basically toggles `preserveModules`.

Reply:
I hear you. I had it as `compat` initially, but I changed it to `declaration` because it matches the [tsconfig option](https://www.typescriptlang.org/tsconfig#declaration).

Reply:
Changed to `preserve`